File: chap12/run-model12-8.R
library(cmdstanr)

Y <- read.csv('input/data-1D.csv')$Y
I <- length(Y)
Mu <- rep(mean(log(ifelse(Y == 0, 1, Y))), I)
X <- (1:I - 1) / (I-1)
data <- list(N=I, X=X, Mu=Mu, Y=Y)

model <- cmdstan_model('model/model12-8.stan')
fit <- model$sample(data=data, seed=123, parallel_chains=4)
fit$save_object(file='output/result-model12-8.RDS')

File: chap12/run-model12-8.py
import pandas
import cmdstanpy
import numpy as np

Y = pandas.read_csv('input/data-1D.csv').Y
I = len(Y)
Mu = np.repeat(np.mean(np.log(Y.clip(1))), I)
X = np.arange(I) / (I-1)
data = {'N':I, 'X':X, 'Mu':Mu, 'Y':Y}

model = cmdstanpy.CmdStanModel(stan_file='model/model12-8.stan')
fit = model.sample(data=data, seed=123, parallel_chains=4)
fit.save_csvfiles('output/result-model12-8')
